posthog-core/src/index.ts

```diff
@@ -839,10 +840,11 @@ export abstract class PostHogCoreStateless {
 
     const payload = JSON.stringify(data)
 
+    const canGzip = this.captureMode === 'json' && isGzipSupported();
```

Comment:
we can call `isGzipSupported` once when the SDK is started instead of on every single flush.
we should also offer a way of disabling `gzip` even if its possible.